app/src/main/java/com/limelight/binding/input/trackpad/Core/TrackpadFramework.java

```diff
@@ -0,0 +1,786 @@
+    private static final short clickToDragThreshold = 200;
+    private static final short dragToDoubleClickThreshold = 200;
+
+    private Timer buttonTimer;
+
+    private static short autoMoveSpaceThreshold = 100;
+
+    public TrackpadFramework(TrackpadListener listener)
+    {
+        trackpadListener = listener;
+        String deviceHead = Build.MODEL.substring(0,5);
+
+        Log.i(TAG,"DeviceName : "+ deviceHead);
+
+        if("SM-X9".equals(deviceHead)){ // Tab S8 Ultra
+            maxXThreshold = 1763;
+            minXThreshold = 0;
+            maxYThreshold = 1599;
+            minYThreshold = 528;
+            autoMoveSpaceThreshold = 100;
+        }else if("SM-T9".equals(deviceHead)||"SM-X8".equals(deviceHead)){ // Tab S7-S8 PLUS
+            maxXThreshold = 1559;
+            minXThreshold = 0;
+            maxYThreshold = 1599;
+            minYThreshold = 780;
+            autoMoveSpaceThreshold = 100;
+        }else {
+            Log.i(TAG,"Not listed : Calibration is required to use AutoMove.");
+            maxXThreshold = minXThreshold = maxYThreshold = minYThreshold = autoMoveSpaceThreshold = -1;
+        }
+    }
+
+    public boolean trackpadHandler(MotionEvent motionEvent){
+        pointerManager(motionEvent);
+        switch (pointerMode){
+            case 0:
+                onePointer(motionEvent);
+                break;
+            case 1:
+                doublePointer(motionEvent);
+                break;
+            case 2:
+                triplePointer(motionEvent);
+                break;
+        }
+        return true;
+    }
+
+    @TargetApi(Build.VERSION_CODES.Q)
+    private float[][] axisManager(MotionEvent motionEvent){
+        float[][] raw = new float[2][5];
+        for(int i = 0; i< motionEvent.getPointerCount(); i++){
+            raw[0][i] = motionEvent.getRawY(i)-minXThreshold;
+            raw[1][i] = motionEvent.getRawX(i)-minYThreshold;
+        }
+        return raw;
+    }
+
+
+    private int maxXThreshold;
+    private int minXThreshold;
+    private int maxYThreshold;
+    private int minYThreshold;
+
+    private void trackPadCalibration(MotionEvent motionEvent){
+        float x = motionEvent.getRawY();
+        float y = motionEvent.getRawX();
+        maxXThreshold = minXThreshold = maxYThreshold = minYThreshold = 1000;
+        if(maxXThreshold <x|| maxYThreshold <y|| minXThreshold >x|| minYThreshold >y){
+            if(maxXThreshold <x) maxXThreshold = (int) x;
+            if(maxYThreshold <y) maxYThreshold = (int) y;
+            if(minXThreshold >x) minXThreshold = (int) x;
+            if(minYThreshold >y) minYThreshold = (int) y;
+        }
+    }
+
+    // Time
+    private final long[] actionDownTime = new long[4];
+    private final long[] actionUpTime = new long[4];
+
+    private int pointerMode=0;
+
+    private void pointerManager(MotionEvent motionEvent){
+        long upTime = SystemClock.uptimeMillis();
```

Comment:
Use `motionEvent.getEventTime()` instead of `SystemClock.uptimeMillis()`

Reply:
Ok. I'll change it.